src/soroban/parsers.ts

```diff
@@ -23,12 +23,17 @@ export function parseRawEvents(
   return {
     latestLedger: r.latestLedger,
     events: (r.events ?? []).map((evt) => {
-      return {
-        ...evt,
-        contractId: new Contract(evt.contractId),
+      // the contractId may be empty so we omit the field in that case
+      const cid = evt.contractId;
+      delete (evt as any).contractId;
+      const rv = {
+        ...evt as Omit<Api.RawEventResponse, 'contractId'>,
+        ...(cid !== '' && { contractId: new Contract(cid) }),
         topic: evt.topic.map((topic) => xdr.ScVal.fromXDR(topic, 'base64')),
         value: xdr.ScVal.fromXDR(evt.value.xdr, 'base64')
       };
+      evt.contractId = cid; // put value back after parsing
```

Comment:
can the side effect on raw evt be avoided with:

```
events: (r.events ?? []).map((evt) => {
      // the contractId may be empty so we omit the field in that case
      const cid = evt.contractId;
      const eventRequiredFields = evt as Omit<Api.RawEventResponse, 'contractId'> 
      const rv = {
        ...eventRequiredFields,
        ...(cid !== '' && { contractId: new Contract(cid) }),
        topic: evt.topic.map((topic) => xdr.ScVal.fromXDR(topic, 'base64')),
        value: xdr.ScVal.fromXDR(evt.value.xdr, 'base64')
      };
      return rv;
    })
```

Reply:
that'd be awesome, but unfortunately the `as Omit` is basically just a typecast rather than actually removing the field from `evt` :cry:

Reply:
ah ok, subtle..is it worth cloning(shallow) to avoid side effect stuff:
```
const eventRequiredFields = {...evt};
delete (eventRequiredFields as any).contractId;
```

thanks for considerations!

Reply:
yeah a general clone is cleaner, you're right! done in a836c990